File: Java/AlarmsProcedural/src/main/java/Main/Alarm.java
package Main;

import Enums.AlarmTypes;
import Enums.TimeCriteria;

public class Alarm {
  public Alarm(AlarmTypes alarmTypes) {
    this.alarmTypes = alarmTypes;
  }

  public AlarmTypes alarmTypes;

  public Alarm nested1;
  public Alarm nested2;
  public TimeCriteria[] timeCriterias;
  public String numberToCall;
}

File: TellDontAsk/AlarmsProcedural/Alarm.cs
using AlarmsProcedural.Enums;

namespace AlarmsProcedural
{
  public class Alarm
  {
    public Alarm(AlarmTypes alarmType)
    {
      AlarmType = alarmType;
    }

    public AlarmTypes AlarmType
    {
      get; set; 
    }

    public Alarm Nested1 { get; set; }
    public Alarm Nested2 { get; set; }
    public TimeCriteria[] TimeCriteria { get; set; }
    public string NumberToCall { get; set; }
  }
}